src/k8s-extension/azext_k8s_extension/partner_extensions/DefaultExtension.py

```diff
@@ -56,5 +58,11 @@ def Update(self, extension, auto_upgrade_minor_version, release_train, version):
             version=version
         )
 
-    def Delete(self, client, resource_group_name, cluster_name, name, cluster_type):
-        pass
+    def Delete(self, cmd, client, resource_group_name, cluster_name, name, cluster_type, yes):
+        user_confirmation_factory(cmd, yes)
+
+
+def user_confirmation_factory(cmd, yes, message="Are you sure you want to perform this operation?"):
+    if cmd.cli_ctx.config.getboolean('core', 'disable_confirm_prompt', fallback=False):
+        return
+    user_confirmation(message, yes=yes)

```

Comment:
Why do we need a config to turn off confirmation?

Reply:
This is an azure core config concept. If that someone has done `az config set core.disable_confirm_prompt=true`, we want to honor this as well for the user_confirmations that we create